### PR TITLE
Persist oidc env vars in deploy template

### PR DESCRIPTION
**Pull request by benbp**
Example usage: https://github.com/Azure/azure-sdk-for-go/pull/23549

### CODE REVIEW

File: eng/common/TestResources/deploy-test-resources.yml [hunk 1]
```diff
@@ -41,12 +42,33 @@ steps:
...
+        ARM_CLIENT_ID: $(ARM_CLIENT_ID)
+        ARM_TENANT_ID: $(ARM_TENANT_ID)
+      inputs:
+        azureSubscription: ${{ parameters.ServiceConnection }}
+        addSpnToEnvironment: true
+        scriptLocation: inlineScript
+        scriptType: pscore
+        inlineScript: |
+          Write-Host "##vso[task.setvariable variable=ARM_OIDC_TOKEN;issecret=true]$($env:idToken)"
+          Write-Host "##vso[task.setvariable variable=ARM_CLIENT_ID;issecret=true]$($env:servicePrincipalId)"
```
Comment (weshaggard) [inline eng/common/TestResources/deploy-test-resources.yml:59]: Do we need these others given we have them available already?

**Review by weshaggard — APPROVED** [review]
We should scope to just OIDC token if possible.

File: eng/common/TestResources/deploy-test-resources.yml (hunk 1 above)
Comment (benbp) [reply eng/common/TestResources/deploy-test-resources.yml:59]: Oh I wasn't sure. I know that go is only using the oidc token, whereas net had all three set.

Comment (benbp) [reply eng/common/TestResources/deploy-test-resources.yml:59]: I think I already replaced the pre/post scripts to use the psboundparameters anyway. So I'll remove.

**Pull request merged**